pallets/pools/src/lib.rs

```diff
@@ -1287,15 +1458,15 @@ pub mod pallet {
 			// Calculate the new total asset value for each tranche
 			// This uses the current state of the tranches, rather than the cached epoch-close-time values.
 			let mut total_assets = total_assets;
-			let tranche_assets = execution
+			let tranche_assets = executed_amounts
 				.iter()
 				.zip(&mut pool.tranches)
-				.map(|((supply, redeem), tranche)| {
+				.map(|((invest, redeem), tranche)| {
 					Self::update_tranche_debt(tranche)?;
 					tranche
```

Comment:
I am not sure if early return of `Self::update_tranche_debt` does what we expect... Have to look at it in more detail, but what would happen if after updating some tranches, we early return as the update failed?

Reply:
We error out with `Overflow` but have already altered the previous tranches of the `&mut pool` we pass in.

Reply:
This is fine, updating tranche debt should actually be a public & permissionless method. In theory it could be called every single block for every single tranche.